### PR TITLE
fix two issues with handling of processing dependent archives

### PR DESCRIPTION
**Pull request by diosmosis**
### Description:

Bugs fixed:

1) setting whether an archiving request is the root one or not as an instance property of Parameters does not work because when querying archive data for child periods, we will create entirely new instances (in CoreAdminHome\API::prepareArchive()). So instead, use a static variable. This properly prevents recursion here.

2) fix edge case where Goals initiates archiving far more than necessary when there are no conversions (so no data in goals reports) but there are visits (so archiving will still be executed). This was always an issue, but is is far more visible now since we aggregate one blob at a time, and thus query archive IDs once per report. So for every Goals report, we'd archive and try to process dependent archives, even though there are no conversions. And since there is no report data, nothing will be inserted, and we'll try to archive again when we move on to the next report.

### Review

* [ ] [Functional review done](https://developer.matomo.org/guides/pull-request-reviews#functional-review-done)
* [ ] [Potential edge cases thought about](https://developer.matomo.org/guides/pull-request-reviews#potential-edge-cases-thought-about) (behavior of the code with strange input, with strange internal state or possible interactions with other Matomo subsystems)
* [ ] [Usability review done](https://developer.matomo.org/guides/pull-request-reviews#usability-review-done) (is anything maybe unclear or think about anything that would cause people to reach out to support)
* [ ] [Security review done](https://developer.matomo.org/guides/security-in-piwik#checklist)
* [ ] [Wording review done](https://developer.matomo.org/guides/pull-request-reviews#translations-wording-review-done)
* [ ] [Code review done](https://developer.matomo.org/guides/pull-request-reviews#code-review-done)
* [ ] [Tests were added if useful/possible](https://developer.matomo.org/guides/pull-request-reviews#tests-were-added-if-usefulpossible)
* [ ] [Reviewed for breaking changes](https://developer.matomo.org/guides/pull-request-reviews#reviewed-for-breaking-changes)
* [ ] [Developer changelog updated if needed](https://developer.matomo.org/guides/pull-request-reviews#developer-changelog-updated-if-needed)
* [ ] [Documentation added if needed](https://developer.matomo.org/guides/pull-request-reviews#documentation-added-if-needed)
* [ ] Existing documentation updated if needed

### CODE REVIEW

**Comment by sgiehl** [comment]
@diosmosis looks like various tests started to fail. Might be good having those fixed before a review.

**Comment by diosmosis** [comment]
php tests should be passing, UI test failures seem unrelated

File: core/ArchiveProcessor/Parameters.php [hunk 1]
```diff
@@ -270,29 +270,6 @@ public function logStatusDebug()
...
-    /**
-     * Returns `true` if these parameters are part of an initial archiving request.
-     * Returns `false` if these parameters are for an archiving request that was initiated
-     * during archiving.
-     *
-     * @return bool
-     */
-    public function isRootArchiveRequest()
-    {
-        return $this->isRootArchiveRequest;
```
Comment (sgiehl) [inline core/ArchiveProcessor/Parameters.php:282]: The class property still exists, but could be removed now.

**Review by sgiehl — CHANGES_REQUESTED** [review]
Left a minor comment. Besides that the code looks fine.

**Review by sgiehl — APPROVED** [review]

**Pull request merged**